### PR TITLE
feat: add docker mode

### PR DESCRIPTION
**Pull request by avatsaev**
# Related Issue/Addition to code

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Proposed Changes
- Run the app inside a docker container with full automation

# Additional Info
- this will allow people to run, automate and orchestrate a fleet of spam removers on compute clusters (Docker/Kubernetes), or in the cloud (for example AWS EKS).

### CODE REVIEW

**Comment by avatsaev** [comment]
hey @ThioJoe I almost finished Dockerizing your spam remover (great job btw), in case you're not familiar with Docker, this will allow people to run, automate and orchestrate a fleet of spam removers on compute clusters (Docker/Kubernetes), or in the cloud (for example AWS EKS).  

But in order for that to work, the config file must allow full automation, which means zero user input necessary with the config file once the app start.

I've played around with the config file but couldn't find a way to fully automate the runs, the script asks for user input once the scanning is over, is there a way to automate with the config file the way the script deals with scan results?

File: .gitignore [hunk 1]
```diff
@@ -24,6 +24,7 @@ wheels/
 .installed.cfg
 *.egg
 MANIFEST
+client_secrets.json
```
Comment (JorgenPhi) [inline .gitignore:27]: client_secrets.json is already listed in the gitignore

Comment (avatsaev) [reply .gitignore]: removed

**Comment by ThioJoe** [comment]
> hey @ThioJoe I almost finished Dockerizing your spam remover (great job btw), in case you're not familiar with Docker, this will allow people to run, automate and orchestrate a fleet of spam removers on compute clusters (Docker/Kubernetes), or in the cloud (for example AWS EKS).
> 
> But in order for that to work, the config file must allow full automation, which means zero user input necessary with the config file once the app start.
> 
> I've played around with the config file but couldn't find a way to fully automate the runs, the script asks for user input once the scanning is over, is there a way to automate with the config file the way the script deals with scan results?

Ok cool, I'll look this over then merge it.

The config file _should_ allow fully autonomous running, but I may have added a feature here or there and forgot to account for it. One to make sure you enable is the auto_close setting under general

**Comment by JorgenPhi** [comment]
@avatsaev You'd also probably want to add a dockerignore file, so that any files in the local development environment don't get added to the image file. Would not want to accidentally publish an image with tokens / secrets in there.

**Comment by avatsaev** [comment]
Awesome

The only caveat is the script must be run at least once with the user_secrets.json before running Docker compose because the script requires user input to generate a token, so the token must be generated before running docker or docker-compose 

I'll update the readme with instructions or would you prefer a separate documentation file?

@JorgenPhi good idea

**Comment by avatsaev** [comment]
@ThioJoe thanks for merging but please make sure the config file allows for full automation, otherwise this is useless

**Comment by ThioJoe** [comment]
> @ThioJoe thanks for merging but please make sure the config file allows for full automation, otherwise this is useless and will just confuse people.

Could you file a bug report? It will help me track down the circumstances where it's not behaving right

**Comment by avatsaev** [comment]
Played with the config again, turns out I forgot to set logging options in the config file.

Everything works fine in full automation 👍🏻

**Comment by oscrx** [comment]
Running the container in [interactive/tty](https://docs.docker.com/engine/reference/commandline/run/#examples) mode enables user interaction and doesn't require installing the scripts and requirements. This can also be added to the [docker compose](https://docs.docker.com/compose/compose-file/compose-file-v3/#domainname-hostname-ipc-mac_address-privileged-read_only-shm_size-stdin_open-tty-user-working_dir) file.
```yaml
stdin_open: true
tty: true
```

Maybe publishing an image on the Github container registry or docker hub when bundling new versions will also solve version issues for end users.

**Comment by oscrx** [comment]
I proposed some improvements to this in PR #467

**Pull request merged**